survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/LocationPickerPart.kt

```diff
@@ -0,0 +1,268 @@
+    private var search: AutoCompleteTextView
+    private val mapView: MapView
+    private lateinit var map: GoogleMap
+    private val frame = FrameLayout(context)
+    private val locationMarker = ImageView(context)
+
+    private val latitude: Double?
+        get() {
+            return if (::map.isInitialized) {
+                map.cameraPosition.target.latitude
+            } else {
+                null
+            }
+        }
+
+    private val longitude: Double?
+        get() {
+            return if (::map.isInitialized) {
+                map.cameraPosition.target.longitude
+            } else {
+                null
+            }
+        }
+
+    private lateinit var mapReadySelection: Selection
+
+    //endregion
+
+    //region Public API
+
+    var selected: Selection
+        get() = Selection(
+            latitude, longitude
+        )
+        set(selected) {
+            if (::map.isInitialized && selected.longitude != null && selected.latitude != null) {
+                map.moveCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        LatLng(
+                            selected.longitude,
+                            selected.latitude
+                        ),
+                        15f
+                    )
+                )
+            } else {
+                mapReadySelection = selected
+            }
+        }
+
+    //endregion
+
+    //region Overrides
+
+    override fun style(surveyTheme: SurveyTheme) {}
+
+    //endregion
+
+    //region Subtypes
+
+    data class Selection(val latitude: Double?, val longitude: Double?)
+
+    //endregion
+
+    init {
+        this.orientation = VERTICAL
+        this.gravity = Gravity.CENTER
+
+        mapView = MapView(
+            context,
+            GoogleMapOptions().scrollGesturesEnabled(true)
+                .tiltGesturesEnabled(true).rotateGesturesEnabled(false)
+        )
+
+        search = AutoCompleteTextView(context)
+        search.setHint(android.R.string.search_go)
+        search.setSingleLine(true)
+        search.setCompoundDrawables(
+            resources.getDrawable(R.drawable.ic_baseline_search_24),
+            null,
+            null,
+            null
+        )
+        search.compoundDrawablePadding = 32
+
+        mapView.getMapAsync { map ->
+            this.map = map
+            if (::mapReadySelection.isInitialized) {
+                selected = mapReadySelection
+            }
+
+            if (ActivityCompat.checkSelfPermission(
+                    mapView.context,
+                    Manifest.permission.ACCESS_FINE_LOCATION
+                ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+                        mapView.context,
+                        Manifest.permission.ACCESS_COARSE_LOCATION
+                    ) != PackageManager.PERMISSION_GRANTED
+            ) {
+                // TODO ask location permission
+                return@getMapAsync
```

Comment:
To be honest, I think here we should throw. Calling this Question type without the necessary permissions does not make any sense.
Another option: what we could do, is show an empty screen, no maps, with just "Please request xy permission beforehand".

The permissions itsef should be requested from the handling app, not us

Reply:
I agree with you. It can be more complicated before the user answer question. I will remove.